services/ingestor/src/main/kotlin/company/thewebhook/ingestor/Application.kt

```diff
@@ -74,27 +83,34 @@ fun Application.module() = launch {
                                 serverHost,
                             )
                         )
-                    try {
-                        val res = producer.publish("$serverHost-incoming", webhookData)
-                        call.respond(
-                            if (res) {
-                                HttpStatusCode.OK
-                            } else {
-                                call.application.log.error("Publishing webhook data returned false")
-                                HttpStatusCode.InternalServerError
-                            }
-                        )
-                    } catch (e: Exception) {
-                        call.application.log.error("Exception thrown on publishing webhook data", e)
-                        when (e) {
-                            is MessageTooLargeException -> {
-                                call.respond(HttpStatusCode.PayloadTooLarge)
-                            }
-                            else -> {
-                                call.respond(HttpStatusCode.InternalServerError)
+
+                    if(serverHostRegex.matches(serverHost)) {
+                        try {
+                            val res = producer.publish("$serverHost-incoming", webhookData)
+                            call.respond(
+                                if (res) {
+                                    HttpStatusCode.OK
+                                } else {
+                                    call.application.log.error("Publishing webhook data returned false")
+                                    HttpStatusCode.InternalServerError
+                                }
+                            )
+                        } catch (e: Exception) {
+                            call.application.log.error("Exception thrown on publishing webhook data", e)
+                            when (e) {
+                                is MessageTooLargeException -> {
+                                    call.respond(HttpStatusCode.PayloadTooLarge)
+                                }
+                                else -> {
+                                    call.respond(HttpStatusCode.InternalServerError)
+                                }
                             }
                         }
                     }
+                    else {
+                        call.application.log.error("Invalid server host")
+                        call.respond(HttpStatusCode.Forbidden)
+                    }
```

Comment:
Instead of putting this entire code in an `if`, can u put the current else condition in an if at the top and return immediately?

Reply:
Sure